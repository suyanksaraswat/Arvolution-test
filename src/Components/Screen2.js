import { Button, Row, Col, Layout } from 'antd';
import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';

function Screen2(props) {
    const history = useHistory();

    const player1Name = useState(history.location.state?.player1Name);
    const player2Name = useState(history.location.state?.player2Name);
    const [player1Win, setPlayer1Win] = useState(0);
    const [player2Win, setPlayer2Win] = useState(0);

    useEffect(() => {
        if (!player1Name || player1Name === '' || !player2Name || player2Name === '') {
            history.push('/');
        }
    });

    const handlePlayer1Game = () => setPlayer1Win(player1Win + 1);

    const handlePlayer2Game = () => setPlayer2Win(player2Win + 1);


    return (
        <Layout>
            <Layout.Content>
                <Row className='full-screen-height'>
                    <Col xs={3} sm={3} md={9} />
                    <Col xs={18} sm={18} md={6} className='column-center'>
                        <div className='login-container'>
                            <div style={{display: 'flex', justifyContent: 'space-between'}}>
                                <span>
                                    {player1Name}
                                </span>
                                <Button
                                    onClick={() => handlePlayer1Game()}
                                >
                                    Add Win
                                </Button>
                            </div>
                            <div style={{display: 'flex', justifyContent: 'space-between'}}>
                                <span>
                                    Wins:
                                </span>
                                <span>
                                    {player1Win}
                                </span>
                            </div>
                            <div style={{display: 'flex', justifyContent: 'space-between'}}>
                                <span>
                                    {player2Name}
                                </span>
                                <Button
                                    onClick={() => handlePlayer2Game()}
                                >
                                    Add Win
                                </Button>
                            </div>
                            <div style={{display: 'flex', justifyContent: 'space-between'}}>
                                <span>
                                    Wins:
                                </span>
                                <span>
                                    {player2Win}
                                </span>
                            </div>
                            <div style={{display: 'flex', justifyContent: 'space-between'}}>
                                <span>
                                    Current Winner:
                                </span>
                                <span>
                                {player1Win > player2Win ? player1Name : (player2Win > player1Win ? player2Name : 'Draw')}
                                </span>
                            </div>
                            <div style={{display: 'flex', justifyContent: 'space-between'}}>
                                <span>
                                    Win Difference:
                                </span>
                                <span>
                                    {Math.abs(player1Win - player2Win)}
                                </span>
                            </div>
                        </div>
                    </Col>
                    <Col xs={3} sm={3} md={9} />
                </Row>
            </Layout.Content>
        </Layout>
    );
}
    
export default Screen2;