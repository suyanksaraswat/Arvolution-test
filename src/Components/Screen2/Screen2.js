import { Button, Row, Col, Layout, Divider } from 'antd';
import { ArrowLeftOutlined } from '@ant-design/icons';
import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';

function Screen2(props) {
    const history = useHistory();

    const player1Name = useState(history?.location?.state?.player1Name);
    const player2Name = useState(history?.location?.state?.player2Name);
    const [player1Win, setPlayer1Win] = useState(0);
    const [player2Win, setPlayer2Win] = useState(0);

    useEffect(() => {
        if (!player1Name || player1Name === '' || !player2Name || player2Name === '') {
            history.push('/');
        }
    });

    return (
        <Layout>
            <Layout.Content>
                <Row className='full-screen-height'>
                    <Col xs={3} sm={3} md={9} />
                    <Col xs={18} sm={18} md={6} className='column-center'>
                        <div className='mb-10'>
                            {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
                            <a onClick={() => history.push('/')}>
                                <ArrowLeftOutlined />&nbsp;Back
                            </a>
                        </div>
                        <div className='login-container'>
                            <div className='display-flex justify-space-between mb-10'>
                                <h1 className='playerName'>
                                    {player1Name}
                                </h1>
                                <span className='addWin-btn'>
                                    <Button
                                        onClick={() => setPlayer1Win(player1Win + 1)}
                                    >
                                        Add Win
                                    </Button>
                                </span>
                            </div>
                            <div className='display-flex justify-space-between mb-20'>
                                <h3>
                                    Wins:
                                </h3>
                                <h1>
                                    {player1Win}
                                </h1>
                            </div>
                            <div className='display-flex justify-space-between mb-10'>
                                <h1 className='playerName'>
                                    {player2Name}
                                </h1>
                                <span className='addWin-btn'>
                                    <Button
                                        onClick={() => setPlayer2Win(player2Win + 1)}
                                    >
                                        Add Win
                                    </Button>
                                </span>
                            </div>
                            <div className='display-flex justify-space-between mb-20'>
                                <h3>
                                    Wins:
                                </h3>
                                <h1>
                                    {player2Win}
                                </h1>
                            </div>
                            <Divider type={'horizontal'} />
                            <div className='display-flex justify-space-between mb-10'>
                                <span>
                                    Current Winner:
                                </span>
                                <span className='playerName'>
                                {player1Win > player2Win ? player1Name : (player2Win > player1Win ? player2Name : 'Draw')}
                                </span>
                            </div>
                            <div className='display-flex justify-space-between'>
                                <span>
                                    Win Difference:
                                </span>
                                <span>
                                    {Math.abs(player1Win - player2Win)}
                                </span>
                            </div>
                        </div>
                    </Col>
                    <Col xs={3} sm={3} md={9} />
                </Row>
            </Layout.Content>
        </Layout>
    );
}
    
export default Screen2;