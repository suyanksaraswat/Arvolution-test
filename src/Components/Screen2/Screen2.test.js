import React from 'react';
import ReactDOM from 'react-dom';
import Screen2 from './Screen2';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from "react-test-renderer";

afterEach(cleanup);

it("screen2 renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<Screen2></Screen2>, div)
})

it("matches snapshot 1", () => {
	const tree = renderer.create(<Screen2></Screen2>).toJSON();
	expect(tree).toMatchSnapshot();
})