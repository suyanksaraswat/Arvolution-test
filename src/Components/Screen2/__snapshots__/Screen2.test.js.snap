// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<section
  className="ant-layout"
>
  <main
    className="ant-layout-content"
  >
    <div
      className="ant-row full-screen-height"
      style={Object {}}
    >
      <div
        className="ant-col ant-col-xs-3 ant-col-sm-3 ant-col-md-9"
        style={Object {}}
      />
      <div
        className="ant-col column-center ant-col-xs-18 ant-col-sm-18 ant-col-md-6"
        style={Object {}}
      >
        <div
          className="mb-10"
        >
          <a
            onClick={[Function]}
          >
            <span
              aria-label="arrow-left"
              className="anticon anticon-arrow-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="arrow-left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
            Â Back
          </a>
        </div>
        <div
          className="login-container"
        >
          <div
            className="display-flex justify-space-between mb-10"
          >
            <h1
              className="playerName"
            />
            <span
              className="addWin-btn"
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Add Win
                </span>
              </button>
            </span>
          </div>
          <div
            className="display-flex justify-space-between mb-20"
          >
            <h3>
              Wins:
            </h3>
            <h1>
              0
            </h1>
          </div>
          <div
            className="display-flex justify-space-between mb-10"
          >
            <h1
              className="playerName"
            />
            <span
              className="addWin-btn"
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Add Win
                </span>
              </button>
            </span>
          </div>
          <div
            className="display-flex justify-space-between mb-20"
          >
            <h3>
              Wins:
            </h3>
            <h1>
              0
            </h1>
          </div>
          <div
            className="ant-divider ant-divider-horizontal"
            role="separator"
          />
          <div
            className="display-flex justify-space-between mb-10"
          >
            <span>
              Current Winner:
            </span>
            <span
              className="playerName"
            >
              Draw
            </span>
          </div>
          <div
            className="display-flex justify-space-between"
          >
            <span>
              Win Difference:
            </span>
            <span>
              0
            </span>
          </div>
        </div>
      </div>
      <div
        className="ant-col ant-col-xs-3 ant-col-sm-3 ant-col-md-9"
        style={Object {}}
      />
    </div>
  </main>
</section>
`;
